### PR TITLE
add wofz (Faddeeva function) and extend erfcx to complex inputs

### PR DESCRIPTION
**Pull request by TheSkyentist**
## Summary

Adds jax.scipy.special.wofz, the Faddeeva function $w(z) = e^{-z^2}\mathrm{erfc}(-iz)$, using the Weideman (1994) N=32 rational approximation. Also extends jax.scipy.special.erfcx to accept complex inputs by routing them through wofz.

Continuing to add to this family of functions from #37580.

## Implementation:
  - _wofz_upper(z): evaluates the rational approximation via jnp.polyval for Im(z) ≥ 0
  - _wofz(z): applies the reflection formula $w(-z) = 2e^{-z^2} - w(z)$ for the lower half-plane; decorated with @custom_jvp implementing $w'(z) = -2z,w(z) + \frac{2i}{\sqrt{\pi}}$
  - wofz(z): public wrapper promoting real inputs to complex before dispatching
  - erfcx complex branch: erfcx(z) = wofz(iz), matching SciPy behaviour
  - Coefficients stored in descending-degree order to avoid a [::-1] reversal on every polyval call

## Tests added:
  - testWofzAccuracy: float32 accuracy against SciPy reference
  - testWofzLowerHalfPlane: lower-half-plane reflection formula
  - testWofzJvp: JVP correctness against the analytic derivative
  - testErfcxComplex: erfcx(z) == wofz(iz) identity + SciPy reference

### CODE REVIEW

**Comment by TheSkyentist** [comment]
I've addressed the code review comments :-)

File: jax/_src/scipy/special.py [hunk 1]
```diff
@@ -932,6 +980,30 @@ def rel_entr(
...
+
+  Returns:
+    array of complex values of the Faddeeva function.
+
+  See also:
+    - :func:`jax.scipy.special.erfcx`
+  """
+  z, = promote_args_inexact("wofz", z)
+  if dtypes.issubdtype(z.dtype, np.floating):
+    z = z.astype(np.result_type(z.dtype, np.complex64))
```
Comment (jakevdp) [inline jax/_src/scipy/special.py]: We should add a `promote_args_complex` utility in `jax._src.numpy.util` and call it here.

File: tests/lax_scipy_special_functions_test.py [hunk 2]
```diff
@@ -182,6 +182,10 @@ def op_record(name, nargs, dtypes, rng_factory, test_grad, nondiff_argnums=(), t
     ),
     op_record("log_softmax", 1, float_dtypes, jtu.rand_default, True),
     op_record("softmax", 1, float_dtypes, jtu.rand_default, True),
+    op_record(
+        "wofz", 1, jtu.dtypes.complex, jtu.rand_default, False,
+        test_name="wofz_complex"
```
Comment (jakevdp) [inline tests/lax_scipy_special_functions_test.py]: I would drop the test name here – the default should be fine.

File: tests/lax_scipy_special_functions_test.py
Comment (jakevdp) [inline tests/lax_scipy_special_functions_test.py:96]: Should we update this to `float_dtypes + complex_dtypes`?

**Review by jakevdp — CHANGES_REQUESTED** [review]
Thanks for looking into this! A few suggestions below.

**Comment by TheSkyentist** [comment]
I have made the suggested changes, thanks for the eyes on this!

File: tests/lax_scipy_special_functions_test.py [hunk 3]
```diff
@@ -248,6 +249,43 @@ def testErfcxLargeX(self):
...
+  def testWofzJvp(self):
+    # d/dz w(z) = -2z*w(z) + 2i/sqrt(pi) — test against numerical diff.
+    rng = jtu.rand_default(np.random.RandomState(2))
+    z = rng((10,), np.complex64) + 0.5j  # stay in upper half-plane
+    import jax
+    primals, tangents = jax.jvp(lsp_special.wofz, (z,), (np.ones_like(z),))
+    expected_tangents = -2 * z * primals + jnp.array(2j / np.sqrt(np.pi), dtype=z.dtype)
+    self.assertAllClose(tangents, expected_tangents, rtol=1e-4)
+
+  def testErfcxComplex(self):
```
Comment (jakevdp) [inline tests/lax_scipy_special_functions_test.py]: I think since we now cover complex values in the `op_record` tests, this special test case is not necessary.

**Review by jakevdp — CHANGES_REQUESTED** [review]
Thanks! One last comment.

Also, once you address that, could you please squash your changes into a single commit for merge? Thanks!

**Comment by TheSkyentist** [comment]
Good catch! redundant test is now removed.

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!
